en/services/ftp.md

```diff
@@ -404,24 +404,57 @@ The GSC should create a timeout after the `RemoveDirectory` command is sent and
-  Ends when we hit EOF
-  kCmdBurstReadFile
-  
+<!-- Original Diagram
+sequenceDiagram;
+    participant Client
+    participant Server
+    Note right of Client: Open file for burst reading
+    Client->>Server:  BurstReadFile( data[0]=path, size=len(path) )
+    Server-- >>Client: ACK( session, size=4, data=len(file) )
+    Note left of Server: Server streams data in chunks with burst_complete=0
+    Server-- >>Client: BurstReadFile(session,burst_complete=0, offset=0, size=len(buffer), data[0]=buffer)
+    Server-- >>Client: BurstReadFile(session,burst_complete=0, offset=n, size=len(buffer), data[0]=buffer)
+    Note left of Server: When complete, send last chunk with burst_complete=1
+    Server-- >>Client: BurstReadFile(session,burst_complete=1, offset=last_chunk, size=len(buffer), data[0]=buffer)
+    Note right of Client: Read any missing chunks using ReadFile at offset
+    Client- >>Server:  ReadFile(session, size, offset)
+    Server-- >>Client: ACK(session, size=len(buffer), data[0]=buffer)
+    Note right of Client: Close session when whole data file collected
+    Client- >>Server:  TerminateSession(session)
+    Server-- >>Client: ACK()
 -->
 
+The sequence of operations is:
+1. Client (i.e. GCS) sends [BurstReadFile](#BurstReadFile) command specifying the file path to open.
+   - The payload must specify: `data[0]`= file path string, `size`=length of file path string.
+1. Server (drone) responds with either 
+   - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = length of file that has been opened. 
+   - NAK with [error information](#error_codes), e.g. `NoSessionsAvailable`, `FileExists`. 
+     The client may cancel the operation, depending on the error.
+1. Server sends stream of [BurstReadFile](#BurstReadFile) commands to client (without ack) until whole file sent.
+   - The payload must specify: `session`=current session, `size`=size of data to read, `offset`= position in data to start reading for current chunk.
+   - The payload must specify `burst_complete=0` for all chunks, except the last one, which must set `burst_complete=1`.
+   
+   > **Note** The client must maintain its own record of the received (and missing) chunks.
+1. Client sends [ReadFile](#ReadFile) to request any/each of the missing chunks
+   - The payload must specify: `session`=current session, `size`=size of data to read, `offset`= position in data to start reading for chunk.
+1. Server responds with:
+   - ACK on success. The [payload](#payload) fields are: `data` = data chunk requested, `size` = size of data in the `data` field.
+   - NAK on failure with [error information](#error_codes).
+1. Client sends [TerminateSession](#TerminateSession) to close the file once all the chunks have been downloaded
+   The server should send an ACK/NAK, but this may (generally speaking) be ignored by the client.
+
+The client should create a timeout while waiting for a new `BurstReadFile` <!-- and do? -->
```

Comment:
@bkueng What should the client do if it stops getting BurstReadFile chunks? My assumption is that it has a timeout, and then just starts the normal readfile process, but it could do something else.

Reply:
Yes you can do either one: BurstReadFile or ReadFile

Reply:
I understand how this would work for ReadFile - you just send it and wait for ack. But in the BurstReadFile case are you saying that the server just kicks off another whole new session etc? 
Or to put it another way, ReadFile is something you do within a session, but BurstReadFile creates a session - the client only sends it once normally to start a session off - so what would it do in the "restart burst" case caused by an offset?

Does the question make sense?

Reply:
No, BurstReadFile does not create a session. It's very much like ReadFile, just gets you more data.

Reply:
Yes